streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractKeyValueStoreTest.java

```diff
@@ -383,13 +394,29 @@ public void shouldThrowNullPointerExceptionOnGetNullKey() {
     }
 
     @Test
-    public void shouldThrowNullPointerExceptionOnRangeNullFromKey() {
-        assertThrows(NullPointerException.class, () -> store.range(null, 2));
+    public void shouldReturnValueOnRangeNullFromKey() {
+        store.put(0, "zero");
+        store.put(1, "one");
+        store.put(2, "two");
+
+        final LinkedList<KeyValue<Integer, String>> expectedContents = new LinkedList<>();
+        expectedContents.add(new KeyValue<>(0, "zero"));
+        expectedContents.add(new KeyValue<>(1, "one"));
+
+        assertEquals(expectedContents, getOrderedContents(store.range(null, 1)));
     }
 
     @Test
-    public void shouldThrowNullPointerExceptionOnRangeNullToKey() {
-        assertThrows(NullPointerException.class, () -> store.range(2, null));
+    public void shouldReturnValueOnRangeNullToKey() {
```

Comment:
Could we add the same null range tests for `reverseRange` here?

Reply:
Added.